### ISSUE TITLE
Data Explorer strips leading or trailing whitespace from column names.

### ISSUE THREAD

**Issue by jthomasmock** [posit-dev/positron#3089]
Positron Dev Version: 2024.05.0 build 0
Code - OSS Version: 1.89.0
Commit: Unknown
Date: Unknown
Electron: 28.2.8
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Darwin arm64 23.4.0

## Steps to reproduce the issue:

1. Create a dataframe with whitespace in front of the actual name

```r
example <- data.frame(1:10, 1:10, 1:10)

names(example) <- c("age1", "               age2", "age3")
```

```python
import pandas as pd

lead_ws = pd.DataFrame({"          leading_whitespace": [1,2,3], 
                       "trailing whitespace               ": [1,2,3]})
%view lead_ws
```

2. Open it in the Data Explorer

![image](https://github.com/posit-dev/positron/assets/29187501/6329fa72-e2e9-4b22-8d45-6dc225b597bb)

![image](https://github.com/posit-dev/positron/assets/29187501/e9630480-380f-41ff-9112-23906e7a6cf7)

3. Note that all the leading and trailing whitespace has been stripped.

## What did you expect to happen?

We should honor the exact name of the column.

**Comment by wesm** [comment]
Yes, please open a new issue for that. I don't believe that the change I made affects formatting of whitespace in the middle, so if we want to fix that we'll have to do some more work (e.g. using middle dots for anything more than a single space).

**Comment by jonvanausdeln** [comment]
### Verified Fixed
Positron Version(s) : [2025.01.0-152](https://github.com/posit-dev/positron-builds/releases/tag/2025.01.0-152)
OS Version(s) : Windows 11

### Test scenario(s) 
Leading and trailing spaces are preserved as expected.  

### Link(s) to TestRail test cases run or created:
